src/v/kafka/server/handlers/create_topics.cc

```diff
@@ -203,6 +204,38 @@ ss::future<response_ptr> create_topics_handler::handle(
           });
         co_return co_await ctx.respond(std::move(response));
     }
+
+    // Record the number of partition mutations in each requested topic,
+    // calulcating throttle delay if necessary
+    auto quota_exceeded_it = co_await ssx::partition(
+      begin,
+      valid_range_end,
+      [&ctx, &response](const creatable_topic& t) -> ss::future<bool> {
+          const auto delay
+            = co_await ctx.quota_mgr().record_partition_mutations(
+              ctx.header().client_id, t.num_partitions);
+          response.data.throttle_time_ms = std::max(
+            response.data.throttle_time_ms, delay);
```

Comment:
looks like `response` might suffer from UAF since it is used in the coroutine lambda following the co_await?

Reply:
I have changed this method to a co-routine so `response` should be alive in this context

Reply:
the problem is not that the underlying `response` lifetime is incorrect, but that the `response` reference stored in the lambda object is destroyed and accessed after co_await. anything inside of a capture list should not be considered safe to access after a co_await in a lambda coroutine. you can copy them to the stack before co_await or sometimes it is appropriate to use ss::coroutine::lambda.

Reply:
ah thanks for the clarification, fixed